src/pages-and-resources/live/Settings.test.jsx

```diff
@@ -0,0 +1,219 @@
+import {
+  render,
+  act,
+  fireEvent,
+  waitFor,
+} from '@testing-library/react';
+
+import { Switch } from 'react-router-dom';
+import { initializeMockApp, history } from '@edx/frontend-platform';
+import MockAdapter from 'axios-mock-adapter';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import { AppProvider, PageRoute } from '@edx/frontend-platform/react';
+import { IntlProvider } from '@edx/frontend-platform/i18n';
+
+import initializeStore from '../../store';
+import { executeThunk } from '../../utils';
+import LiveSettings from './Settings';
+import {
+  generateLiveConfigurationApiResponse,
+  courseId,
+  initialState,
+} from './factories/mockApiResponses';
+
+import { fetchLiveConfiguration } from './data/thunks';
+import { providerConfigurationApiUrl } from './data/api';
+import messages from './messages';
+import PagesAndResourcesProvider from '../PagesAndResourcesProvider';
+
+let axiosMock;
+let container;
+let store;
+const liveSettingsUrl = `/course/${courseId}/pages-and-resources/live/settings`;
+
+const renderComponent = async () => {
+  const wrapper = await render(
+    <IntlProvider locale="en">
+      <AppProvider store={store}>
+        <PagesAndResourcesProvider courseId={courseId}>
+          <Switch>
+            <PageRoute path={liveSettingsUrl}>
+              <LiveSettings onClose={() => {}} />
+            </PageRoute>
+          </Switch>
+        </PagesAndResourcesProvider>
+      </AppProvider>
+    </IntlProvider>,
+  );
+  container = wrapper.container;
+  return wrapper;
+};
+
+describe('LiveSettings', () => {
+  beforeEach(async () => {
+    initializeMockApp({
+      authenticatedUser: {
+        userId: 3,
+        username: 'abc123',
+        administrator: false,
+        roles: [],
+      },
+    });
+    store = initializeStore(initialState);
+    axiosMock = new MockAdapter(getAuthenticatedHttpClient());
+    history.push(liveSettingsUrl);
+  });
+
+  test('Live Configuration modal is visible', async () => {
+    const wrapper = await renderComponent();
+    expect(wrapper.queryByRole('dialog')).toBeVisible();
+  });
+
+  test('Displays "Configure Live" heading', async () => {
+    const wrapper = await renderComponent();
+    const headingElement = wrapper.queryByTestId('modal-title');
+    expect(headingElement).toHaveTextContent(messages.heading.defaultMessage);
+  });
+
+  test('Displays title, helper and badge when live configuration button is enabled', async () => {
+    const wrapper = await renderComponent();
+    const label = container.querySelector('label[for="enable-live-toggle"]');
+    const helperText = wrapper.queryByTestId('helper-text');
```

Comment:
I think, we do not need the wrapper anymore after loading the container in global container variable.
const wrapper = await renderComponent();
const label = container.querySelector('label[for="enable-live-toggle"]');
const helperText = wrapper.queryByTestId('helper-text');

Restructure
renderComponent();
const label = container.querySelector('label[for="enable-live-toggle"]');
const helperText = queryByTestId(container, 'helper-text')